DoorShuffle.py

```diff
@@ -940,7 +940,9 @@ def find_small_key_door_candidates(builder, start_regions, world, player):
     checked_doors = set()
     for region in start_regions:
         possible, checked = find_key_door_candidates(region, checked_doors, world, player)
-        candidates.extend(possible)
+        for candidate in possible:
+            if candidate not in candidates:
+                candidates.append(candidate)
```

Comment:
A set would simplify this.

Reply:
Sets are bad for choosing randomly due to no order. (Ruins the possibility of repeatable generation.) This is fine, or could use dict for best of both worlds

Reply:
You're right.  I always thought sets behaved the same as dicts, but they don't.  A dict would be better here so it's not O(n^2), but this solution works.